backend/src/schema/repository.ts

```diff
@@ -68,8 +71,10 @@ const resolvers = {
       }))
 
       const branches = await getBranches(repoLocation)
+      const localBranches = await getLocalBranches(repoLocation)
+      const url = args.url
 
-      return { currentBranch, files, branches }
+      return { currentBranch, files, branches, localBranches, url }
```

Comment:
Do we really need the "all" branches anymore? My idea was to replace the branches list in the repoStateQuery with the local branches as it was the original use case. I feel like the it should only support checking out local branches so it doesn't get too messed up with the tracking information. What are your thoughts?

Reply:
Yeah, I have the same idea as you. I thought that maybe leaving the all branches-query in may be beneficial for future development, but now that I think of it I'm not sure what it would be used for. Really we only need to fetch the local branches and the url for this to work.